DataBiosphere/Saturn-UI
# Correct issue with react-focus-lock [SATURN-1226]

The code change responsible for the issue is [here](https://github.com/theKashey/react-focus-lock/compare/v2.1.0...v2.1.1#diff-757809db6137e7d28cf62c382a00aead) - Lock.js lines 64-71 where the focus setting gets deferred in version 2.1.1 whereas it is not in v2.1.0

What happens is:

1. The focus trap is set to return focus when the trapped component unmounts
  a. This action is deferred ( I believe with a 0 timeout)
2. The trapped component unmounts
3. Our modal, which is not part of focus lock, mounts and gains focus
4. The focus trap executes the deferred focus code

This change uses the API's in the modal to gain back focus and properly return focus. I am essentially using the technique to do this in react-focus-lock, but am not thrilled with the solution (adding additional refs, etc...)

I tried a bunch of combinations of the modal and focus lock API, but could not get the desired behavior. This solution does work and gives us the same behavior that currently exists.

Tested by clicking through the application where modals are used and ensuring that setting focus is behaving correctly

To test the before / after behavior:
1. Run npm install on this branch
2. Switch to another branch or dev and open a modal on the workspaces page 
    a. You will see that the modal does not get focus
3. Switch back to this branch and the problem should be resolved
